src/components/MessageBox/messageBox.vue

```diff
@@ -0,0 +1,105 @@
+<template>
+    <v-dialog :persistent="isPersistent">
+        <v-card class="messageBoxContent">
+            <v-card-text>
+                {{ messageText }}
+                <div
+                    v-for="inputItem in inputList"
+                    :key="inputItem"
+                    class="inputContent"
+                >
+                    <p>{{ inputItem }}</p>
+                    <input class="inputField" v-model="inputVal" type="text" />
+                </div>
+            </v-card-text>
+            <v-card-actions>
+                <v-btn
+                    v-for="buttonItem in buttonList"
+                    :key="buttonItem.text"
+                    class="messageBtn"
+                    block
+                    @click="
+                        $emit(
+                            'buttonClick',
+                            buttonItem.emitOption,
+                            circuitItem,
+                            inputVal
+                        )
+                    "
+                >
+                    {{ buttonItem.text }}
+                </v-btn>
+            </v-card-actions>
+        </v-card>
+    </v-dialog>
+</template>
+
+<script lang="ts" setup>
+import { ref } from '@vue/reactivity'
+import { onUpdated } from '@vue/runtime-core'
+const inputVal = ref('')
+
+onUpdated(() => {
+    inputVal.value = 'Untitled-Circuit'
+})
+
+const props = defineProps({
+    messageText: { type: String, default: '' },
+    isPersistent: { type: Boolean, default: false },
+    buttonList: { type: Array, default: undefined },
+    inputList: { type: Array, default: undefined },
```

Comment:
Check default values here, for type Array default should be empty array if required is set to false, any specific reasons for undefined? undefined should rarely be used as values in JS.

Reply:
Fixed !